django/contrib/admindocs/utils.py

```diff
@@ -151,90 +154,185 @@ def replace_metacharacters(pattern):
     )
 
 
-def replace_named_groups(pattern):
-    r"""
-    Find named groups in `pattern` and replace them with the group name. E.g.,
-    1. ^(?P<a>\w+)/b/(\w+)$ ==> ^<a>/b/(\w+)$
-    2. ^(?P<a>\w+)/b/(?P<c>\w+)/$ ==> ^<a>/b/<c>/$
-    3. ^(?P<a>\w+)/b/(\w+) ==> ^<a>/b/(\w+)
-    4. ^(?P<a>\w+)/b/(?P<c>\w+) ==> ^<a>/b/<c>
+def _find_groups(pattern, group_matcher, named):
```

Comment:
@felixxm I'm handling `named`, `unnamed` and `non-capturing` groups inside this function but I'm not sure if this is the best approach so far, any better ideas?